packages/opentelemetry-api/src/diag/consoleLogger.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DiagLogger, DiagLogFunction } from './logger';
+
+const consoleMap: { n: keyof DiagLogger; c: keyof Console }[] = [
+  { n: 'terminal', c: 'error' },
+  { n: 'critical', c: 'error' },
+  { n: 'error', c: 'error' },
+  { n: 'warn', c: 'warn' },
+  { n: 'info', c: 'info' },
+  { n: 'debug', c: 'debug' },
+  { n: 'trace', c: 'trace' },
+  { n: 'forcedInfo', c: 'info' },
+];
+
+/**
+ * A simple Immutable Console based diagnostic logger which will output any messages to the Console.
+ * If you want to limit the amount of logging to a specific level or lower use the
+ * {@link diagLogLevelFilter}
+ */
+export class DiagConsoleLogger implements DiagLogger {
+  constructor() {
+    function _consoleFunc(funcName: keyof Console): DiagLogFunction {
+      return function () {
+        const orgArguments = arguments;
+        if (console) {
+          // Some environments only expose the console when the F12 developer console is open
+          let theFunc = console[funcName];
+          if (!theFunc) {
+            // Not all environments support all functions
+            theFunc = console.log;
+          }
+
+          // One last final check
+          if (theFunc && typeof theFunc === 'function') {
+            return theFunc.apply(console, orgArguments);
+          }
+        }
+      };
+    }
+
+    for (let lp = 0; lp < consoleMap.length; lp++) {
+      const name = consoleMap[lp].n;
+      let consoleFunc = consoleMap[lp].c;
+      if (console && !console[consoleFunc]) {
```

Comment:
Good call, I don't remember why I put this here as it's bogus because the console my be undefined on IE at the point of construction -- will remove the check and just let the above deal with it.